query-service-impl/src/main/java/org/hypertrace/core/query/service/ExecutionContext.java

```diff
@@ -35,14 +38,83 @@
   // while the selectedColumns
   // is a set of column names.
   private final LinkedHashSet<Expression> allSelections;
+  private final Map<String,String> timeSeriesColumnMap;
+  private final Map<String,List<Long>> timeFilterMap;
 
   public ExecutionContext(String tenantId, QueryRequest request) {
     this.tenantId = tenantId;
     this.selectedColumns = new LinkedHashSet<>();
     this.allSelections = new LinkedHashSet<>();
+    this.timeSeriesColumnMap = new HashMap<>();
+    this.timeFilterMap = new HashMap<String,List<Long>>();
+    analyzeForAvgRate(request);
     analyze(request);
   }
 
+  private void analyzeForAvgRate(QueryRequest request){
+    setTimeSeriesColumnMap(request);
+    setTimeFilterMap(request);
+  }
+
+  private void setTimeSeriesColumnMap(QueryRequest request){
+
+    if(request.getGroupByCount() > 0){
+      for(Expression expression : request.getGroupByList()){
+        if(expression.getValueCase() == ValueCase.FUNCTION && expression.getFunction().getFunctionName().equals("dateTimeConvert")) {
+
+          String column = null;
+          String period = null;
+
+          //assuming only one column and one literal (with period in seconds) is present in one dateTimeConvert groupBy
+          for(Expression childExpression : expression.getFunction().getArgumentsList()) {
+            if(childExpression.getValueCase() == ValueCase.COLUMNIDENTIFIER) {
+              column = childExpression.getColumnIdentifier().getColumnName().split("[.]")[0];
```

Comment:
Why are we splitting the column name?

Reply:
column names are present as say SERVICE.start_time / SERVICE.num_calls. So we need only the initial part.

Reply:
Changed the implementation. Not using columns now.